{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<font size = 6>*GMOdetector notebook* </font><br>\n",
    "**Template to analyze a new batch of images** (v.0.7.2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this workflow, images taken with the macroPhor Array dual RGB/hyperspectral imaging platform are analyzed by a workflow in which regression quantifies fluorescent signals in hyperspectral images, deep learning segments RGB images into different tissues, and these datasets are cross-referenced to produce statistics on growth of transgenic callus and shoot."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Experiment ID and description"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-success\">\n",
    "Provide a short description of the experiment in the below box. This should include unique identifier codes for the experiment, along with a short description of genotypes and treatments studied. The timepoint should also be included. </div>"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Dataset (Phase ID and week timepoint): \"ENTER_NOTE\"\n",
    "\n",
    "This notebook comes from a template which we use in R scripts to automatically generate a notebook for each \n",
    "transformation GWAS phase x timepoint."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": "true"
   },
   "source": [
    "# Parameters for analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-success\">\n",
    "The below variables must be modified appropriately every time this workflow is run over new images.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data location\n",
    "The `data` variable below provides the **complete** path to the folder containing data to be analyzed. This should include all folders and subfolders in which the data of interest is organized by. For the organizational system used for our lab's data, this should follow the format \"/Experiment/Subexperiment/Timepoint/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "data=\"ENTER_DATA_PATH\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sample information\n",
    "Every experiment has a spreadsheet of metadata to organize treatment and genotype information for each plate, prepare labels, and randomize plates. [For details, see tutorial on preparing this spreadsheet](https://github.com/naglemi/GMOnotebook/blob/master/1_Decide_parameters/1_Metadata_and_randomization/1-Generate_randomization_scheme.ipynb)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "randomization_datasheet=\"ENTER_RANDOMIZATION_DATASHEET_PATH\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "grid=ENTER_GRID # 12 or 20"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Detection of missing or contaminated explants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set the `missing_explants` variable to `\"Automatic\"` or to the path of manually prepared data file. [For details, see this tutorial and example file](https://github.com/naglemi/GMOnotebook/tree/master/1_Decide_parameters/3_Other_parameters)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: Our automatic missing explant detection model is only trained for poplar."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_explants=\"ENTER_DENSENET_OPTION_OR_SHEET\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Segmentation settings and models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "segmentation_mode=\"ENTER_SEGMENTATION_MODE\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These three below settings need to be changed if you wish to use a different model for hyperspectral segmentation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "segmentation_model_key=\"ENTER_HYP-SEGMENTATION_MODEL_KEY\"\n",
    "segmentation_model_path=\"ENTER_HYP-SEGMENTATION_MODEL_PATH\"\n",
    "segmentation_model_type=\"ENTER_HYP-SEGMENTATION_MODEL_TYPE\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is a list of classes that should not be included in the \"All regenerated tissues\" statistics."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "unregenerated_tissues=\"Background Stem Necrotic Explant\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Computing weights for fluorescent proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[See this notebook for details on all below fluorescent protein settings.](https://github.com/naglemi/GMOnotebook/blob/master/1_Decide_parameters/3_Other_parameters/3_Hyperspectral_settings.ipynb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ALL known fluorescent components in the sample should be included.\n",
    "# Library has DsRed, ZsYellow, GFP, Chl, ChlA, ChlB, Noise\n",
    "fluorophores=ENTER_FLUOROPHORES # Order doesn't matter here. Names must match library.\n",
    "desired_wavelength_range=ENTER_WAVELENGTHS # (first last), e.g. (500 900)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Producing false-color plots for fluorescent proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "FalseColor_channels=ENTER_CHANNELS # (Red Green Blue), e.g. (Chl GFP Noise)\n",
    "FalseColor_caps=ENTER_CAPS # (Red Green Blue); recommend 400 for reporters, 200 for others; e.g. (200 400 200)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Producing summary statistics for fluorescent proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "reporters=ENTER_REPORTERS # Will compute summary stats for these proteins, e.g. (GFP) or (GFP Chl)\n",
    "pixel_threshold=ENTER_PIXEL_THRESHOLD # If this many pixels... (recommended: 3)\n",
    "reporter_threshold=ENTER_REPORTER_THRESHOLD # ...have this much signal (recommended: 38), then the tissue is \"Positive\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "composite=ENTER_COMPOSITE_OPTION # 1 to make composite images with side-by-side RGB, segmentation outputs and blended images (slow), 0 to skip\n",
    "width=ENTER_PLOT_WIDTH # GGplot box/violin plot output (inches)\n",
    "height=ENTER_PLOT_HEIGHT # GGplot box/violin plot output (inches)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parallelization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "parallel=ENTER_PARALLEL_OPTION # 1 if parallelizing CubeGLM with GNU Parallel, 0 if not"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Paths to workflow modules"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These only need to be modified if you are setting up a `GMOnotebook` template in a new environment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "gmodetector_wd=\"/home/cubeglm/\"\n",
    "spectral_library_path=\"${gmodetector_wd}spectral_library/\"\n",
    "deeplab_path=\"/mnt/models/rgb/poplar_model_2_w_contam/\"\n",
    "densenet_model_path=\"/mnt/models/densenet_contamination/model_finetune.h5\"\n",
    "cubeml_path=\"/home/cubeml/\"\n",
    "alignment_path=\"/home/ImageAlignment/\"\n",
    "gmolabeler_path=\"/home/GMOlabeler/\"\n",
    "contamination_path=\"/home/DenseNet\"\n",
    "data_prefix=\"/mnt/output/\"\n",
    "output_directory_prefix=\"${data_prefix}gmodetector_out/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "cwd=\"/home/GMOnotebook\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-info\">\n",
    "With all above variables set, please \"Save as...\" with a filename referencing this specific dataset. <br>Finally, deploy the workflow (Step 4 in above instructions).\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check if inputs are OK"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This script will print warnings for any common problems that are detected with input variables we set above."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GFP Chl Noise\n"
     ]
    }
   ],
   "source": [
    "echo ${fluorophores[@]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GFP[@]\n"
     ]
    }
   ],
   "source": [
    "echo $fluorophores[@]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "opt$data <- \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/\"\n",
      "opt$randomization_datasheet <- \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/GTNEC_labels.xlsx\"\n",
      "opt$segmentation_mode <- \"rgb\"\n",
      "opt$unregenerated_tissues <- \"Background Stem Necrotic Explant\"\n",
      "opt$grid <- \"12\"\n",
      "opt$missing_explants <- \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//output.csv\"\n",
      "opt$fluorophores <- \"GFP,Chl,Noise\"\n",
      "opt$desired_wavelength_range <- \"500,900\"\n",
      "opt$FalseColor_channels <- \"Chl,GFP,Noise\"\n",
      "opt$FalseColor_caps <- \"200,200,200\"\n",
      "opt$reporters <- \"GFP,Chl\"\n",
      "opt$pixel_threshold <- \"3\"\n",
      "opt$reporter_threshold <- \"38\"\n",
      "opt$segmentation_model_key <- \"\"\n",
      "opt$segmentation_model_path <- \"\"\n",
      "opt$gmodetector_wd <- \"/home/cubeglm/\"\n",
      "opt$spectral_library_path <- \"/home/cubeglm/spectral_library/\"\n",
      "opt$deeplab_path <- \"/mnt/models/rgb/poplar_model_2_w_contam/\"\n",
      "opt$cubeml_path <- \"/home/cubeml/\"\n",
      "opt$alignment_path <- \"/home/ImageAlignment/\"\n",
      "opt$gmolabeler_path <- \"/home/GMOlabeler/\"\n",
      "opt$contamination_path <- \"/home/DenseNet\"\n",
      "opt$data_prefix <- \"/mnt/output/\"\n",
      "opt$output_directory_prefix <- \"/mnt/output/gmodetector_out/\"\n",
      "opt$cwd <- \"/home/GMOnotebook\"\n"
     ]
    }
   ],
   "source": [
    "IFS=','  # Setting the Internal Field Separator to ',' for array joining\n",
    "\n",
    "echo \"opt\\$data <- \\\"$data\\\"\"\n",
    "echo \"opt\\$randomization_datasheet <- \\\"$randomization_datasheet\\\"\"\n",
    "echo \"opt\\$segmentation_mode <- \\\"$segmentation_mode\\\"\"\n",
    "echo \"opt\\$unregenerated_tissues <- \\\"$unregenerated_tissues\\\"\"\n",
    "echo \"opt\\$grid <- \\\"$grid\\\"\"\n",
    "echo \"opt\\$missing_explants <- \\\"$missing_explants\\\"\"\n",
    "echo \"opt\\$fluorophores <- \\\"${fluorophores[*]}\\\"\"  # Joining array elements with IFS\n",
    "echo \"opt\\$desired_wavelength_range <- \\\"${desired_wavelength_range[*]}\\\"\"  # Joining array elements with IFS\n",
    "echo \"opt\\$FalseColor_channels <- \\\"${FalseColor_channels[*]}\\\"\"  # Joining array elements with IFS\n",
    "echo \"opt\\$FalseColor_caps <- \\\"${FalseColor_caps[*]}\\\"\"  # Joining array elements with IFS\n",
    "echo \"opt\\$reporters <- \\\"${reporters[*]}\\\"\"  # Joining array elements with IFS\n",
    "echo \"opt\\$pixel_threshold <- \\\"$pixel_threshold\\\"\"\n",
    "echo \"opt\\$reporter_threshold <- \\\"$reporter_threshold\\\"\"\n",
    "echo \"opt\\$segmentation_model_key <- \\\"$segmentation_model_key\\\"\"\n",
    "echo \"opt\\$segmentation_model_path <- \\\"$segmentation_model_path\\\"\"\n",
    "echo \"opt\\$gmodetector_wd <- \\\"$gmodetector_wd\\\"\"\n",
    "echo \"opt\\$spectral_library_path <- \\\"$spectral_library_path\\\"\"\n",
    "echo \"opt\\$deeplab_path <- \\\"$deeplab_path\\\"\"\n",
    "echo \"opt\\$cubeml_path <- \\\"$cubeml_path\\\"\"\n",
    "echo \"opt\\$alignment_path <- \\\"$alignment_path\\\"\"\n",
    "echo \"opt\\$gmolabeler_path <- \\\"$gmolabeler_path\\\"\"\n",
    "echo \"opt\\$contamination_path <- \\\"$contamination_path\\\"\"\n",
    "echo \"opt\\$data_prefix <- \\\"$data_prefix\\\"\"\n",
    "echo \"opt\\$output_directory_prefix <- \\\"$output_directory_prefix\\\"\"\n",
    "echo \"opt\\$cwd <- \\\"$cwd\\\"\"\n",
    "\n",
    "unset IFS  # Resetting IFS back to default\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "In check_existence(opt$segmentation_model_key) : file does not exist: \n",
      "Warning message:\n",
      "In check_existence(opt$segmentation_model_path) : file does not exist: \n",
      "Warning message:\n",
      "In check_spectral_library(opt$fluorophores, opt$spectral_library_path) :\n",
      "  Name not found in spectral library: GFP,Chl,Noise\n",
      "Warning message:\n",
      "In check_spectral_library(opt$FalseColor_channels, opt$spectral_library_path) :\n",
      "  Name not found in spectral library: Chl,GFP,Noise\n",
      "Warning message:\n",
      "In check_spectral_library(opt$reporters, opt$spectral_library_path) :\n",
      "  Name not found in spectral library: GFP,Chl\n",
      "Warning message:\n",
      "NAs introduced by coercion \n",
      "Warning message:\n",
      "desired_wavelength_range must contain two integers \n",
      "Warning message:\n",
      "desired_wavelength_range must contain two integers \n",
      "Warning message:\n",
      "FalseColor_caps must contain three integers \n",
      "\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m•\u001b[39m `` -> `...10`\n",
      "\u001b[?25h\u001b[?25hWarning message:\n",
      "The second column must be a string with column name 'TrayID' \n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25hCheck complete!\n",
      "\u001b[?25h\u001b[?25h\n"
     ]
    }
   ],
   "source": [
    "Rscript ${cwd}/intermediates/are_inputs_ok.R \\\n",
    "  --data \"$data\" \\\n",
    "  --randomization_datasheet \"$randomization_datasheet\" \\\n",
    "  --segmentation_mode \"$segmentation_mode\" \\\n",
    "  --unregenerated_tissues \"$unregenerated_tissues\" \\\n",
    "  --grid \"$grid\" \\\n",
    "  --missing_explants \"$missing_explants\" \\\n",
    "  --fluorophores \"$(IFS=,; echo \"${fluorophores[*]}\")\" \\\n",
    "  --desired_wavelength_range \"$(IFS=,; echo \"${desired_wavelength_range[*]}\")\" \\\n",
    "  --FalseColor_channels \"$(IFS=,; echo \"${FalseColor_channels[*]}\")\" \\\n",
    "  --FalseColor_caps \"$(IFS=,; echo \"${FalseColor_caps[*]}\")\" \\\n",
    "  --reporters \"$(IFS=,; echo \"${reporters[*]}\")\" \\\n",
    "  --pixel_threshold \"$pixel_threshold\" \\\n",
    "  --reporter_threshold \"$reporter_threshold\" \\\n",
    "  --segmentation_model_key \"$segmentation_model_key\" \\\n",
    "  --segmentation_model_path \"$segmentation_model_path\" \\\n",
    "  --gmodetector_wd \"$gmodetector_wd\" \\\n",
    "  --spectral_library_path \"$spectral_library_path\" \\\n",
    "  --deeplab_path \"$deeplab_path\" \\\n",
    "  --cubeml_path \"$cubeml_path\" \\\n",
    "  --alignment_path \"$alignment_path\" \\\n",
    "  --gmolabeler_path \"$gmolabeler_path\" \\\n",
    "  --contamination_path \"$contamination_path\" \\\n",
    "  --data_prefix \"$data_prefix\" \\\n",
    "  --output_directory_prefix \"$output_directory_prefix\" \\\n",
    "  --cwd \"$cwd\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Automated workflow to be deployed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "See the below code for a walkthrough of how GMOnotebook works, or view the outputs after running the workflow for help troubleshooting errors in specific steps of analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-danger\"> <b>Danger:</b> Do not modify any below code without creating a new version of the template notebook. During routine usage, this workflow should be customized only by modifying variables above, while leaving the below code unmodified. </div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These internal variables are set automatically."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "datestamp=$(date +”%Y-%m-%d”)\n",
    "data_folder=$(echo $data | cut -d/ -f5-)\n",
    "timepoint=\"$(basename -- $data_folder)\"\n",
    "output_directory_full=\"$output_directory_prefix$data_folder\"\n",
    "dataset_name=$(echo $data_folder | sed -e 's/\\///g')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "if [ \"$segmentation_mode\" = \"rgb\" ]; then\n",
    "    unset segmentation_model_key\n",
    "    unset segmentation_model_path\n",
    "    unset segmentation_model_type\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Time analysis begins:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sun Dec 24 08:29:55 PM PST 2023\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantification of fluorescent proteins by regression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Python package `CubeGLM` is used to quantify fluorescent proteins in each pixel of hyperspectral images via linear regression. Hyperspectral images are regressed over spectra of known components, and pixelwise maps of test-statistics are constructed for each component in the sample. This approach to quantifying components of hyperspectral images is described in-depth in the Methods section from <a href=\"https://link.springer.com/article/10.1007/s40789-019-0252-7\" target=\"_blank\">Böhme, et al. 2019</a>. Code and documentation for `CubeGLM` is on <a href=\"https://github.com/naglemi/GMOdetector_py\" target=\"_blank\">Github</a>."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data_prefix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_list_name=\"$dataset_name.jobs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -rf $job_list_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in $data/*.hdr\n",
    "do\n",
    " if [[ \"$file\" != *'roadband'* ]]; then\n",
    "  echo \"python -W ignore ${gmodetector_wd}/wrappers/analyze_sample.py \\\n",
    "--file_path $file \\\n",
    "--fluorophores ${fluorophores[*]} \\\n",
    "--min_desired_wavelength ${desired_wavelength_range[0]} \\\n",
    "--max_desired_wavelength ${desired_wavelength_range[1]} \\\n",
    "--red_channel ${FalseColor_channels[0]} \\\n",
    "--green_channel ${FalseColor_channels[1]} \\\n",
    "--blue_channel ${FalseColor_channels[2]} \\\n",
    "--red_cap ${FalseColor_caps[0]} \\\n",
    "--green_cap ${FalseColor_caps[1]} \\\n",
    "--blue_cap ${FalseColor_caps[2]} \\\n",
    "--plot 1 \\\n",
    "--spectral_library_path \"$spectral_library_path\" \\\n",
    "--output_dir $output_directory_full \\\n",
    "--threshold 38\" >> $job_list_name\n",
    " fi\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_Fluorescence.hdr in 29.83200634305831s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_Fluorescence.hdr in 11.905970768071711s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0_Fluorescence.hdr in 11.944245325983502s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100838_1_0_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100838_1_0_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100838_1_0_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100838_1_0_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_100838_1_0_1_Fluorescence.hdr in 11.958818185026757s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100934_2_0_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100934_2_0_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100934_2_0_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100934_2_0_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_100934_2_0_2_Fluorescence.hdr in 11.935619924915954s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101031_3_0_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101031_3_0_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101031_3_0_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101031_3_0_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101031_3_0_3_Fluorescence.hdr in 11.90273538196925s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101127_4_0_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101127_4_0_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101127_4_0_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101127_4_0_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101127_4_0_4_Fluorescence.hdr in 12.3153604590334s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101223_5_0_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101223_5_0_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101223_5_0_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101223_5_0_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101223_5_0_5_Fluorescence.hdr in 11.665446073981002s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101320_6_0_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101320_6_0_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101320_6_0_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101320_6_0_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101320_6_0_6_Fluorescence.hdr in 12.289156455895863s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101416_7_1_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101416_7_1_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101416_7_1_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101416_7_1_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101416_7_1_6_Fluorescence.hdr in 11.672408399987034s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101503_8_1_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101503_8_1_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101503_8_1_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101503_8_1_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101503_8_1_5_Fluorescence.hdr in 12.086786333005875s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101600_9_1_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101600_9_1_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101600_9_1_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101600_9_1_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101600_9_1_4_Fluorescence.hdr in 11.776250411057845s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101656_10_1_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101656_10_1_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101656_10_1_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101656_10_1_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101656_10_1_3_Fluorescence.hdr in 12.08437065104954s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101753_11_1_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101753_11_1_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101753_11_1_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101753_11_1_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101753_11_1_2_Fluorescence.hdr in 11.774173297919333s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101849_12_1_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101849_12_1_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101849_12_1_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101849_12_1_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101849_12_1_1_Fluorescence.hdr in 11.789908451028168s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101946_13_1_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101946_13_1_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101946_13_1_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101946_13_1_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_101946_13_1_0_Fluorescence.hdr in 11.989804919110611s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102042_14_2_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102042_14_2_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102042_14_2_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102042_14_2_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_102042_14_2_0_Fluorescence.hdr in 12.16275294602383s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102129_15_2_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102129_15_2_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102129_15_2_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102129_15_2_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_102129_15_2_1_Fluorescence.hdr in 11.905707375961356s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102225_16_2_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102225_16_2_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102225_16_2_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102225_16_2_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_102225_16_2_2_Fluorescence.hdr in 12.065918226027861s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102322_17_2_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102322_17_2_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102322_17_2_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102322_17_2_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_102322_17_2_3_Fluorescence.hdr in 12.149436107953079s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102418_18_2_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102418_18_2_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102418_18_2_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102418_18_2_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_102418_18_2_4_Fluorescence.hdr in 11.709347367985174s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102515_19_2_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102515_19_2_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102515_19_2_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102515_19_2_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_102515_19_2_5_Fluorescence.hdr in 11.822168458951637s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102611_20_2_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102611_20_2_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102611_20_2_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102611_20_2_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC1_I5.0_F1.9_L100LED30_102611_20_2_6_Fluorescence.hdr in 12.072658254997805s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_102933_0_0_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_102933_0_0_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_102933_0_0_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_102933_0_0_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_102933_0_0_0_Fluorescence.hdr in 12.013558232109062s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103015_1_0_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103015_1_0_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103015_1_0_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103015_1_0_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103015_1_0_1_Fluorescence.hdr in 12.379944672924466s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103112_2_0_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103112_2_0_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103112_2_0_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103112_2_0_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103112_2_0_2_Fluorescence.hdr in 11.865614567999728s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103208_3_0_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103208_3_0_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103208_3_0_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103208_3_0_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103208_3_0_3_Fluorescence.hdr in 12.23249342001509s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103305_4_0_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103305_4_0_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103305_4_0_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103305_4_0_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103305_4_0_4_Fluorescence.hdr in 12.012936815968715s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103401_5_0_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103401_5_0_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103401_5_0_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103401_5_0_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103401_5_0_5_Fluorescence.hdr in 12.043589671025984s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103457_6_0_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103457_6_0_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103457_6_0_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103457_6_0_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103457_6_0_6_Fluorescence.hdr in 11.838664397946559s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103554_7_1_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103554_7_1_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103554_7_1_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103554_7_1_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103554_7_1_6_Fluorescence.hdr in 11.934331018012017s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103641_8_1_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103641_8_1_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103641_8_1_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103641_8_1_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103641_8_1_5_Fluorescence.hdr in 11.968966234009713s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103737_9_1_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103737_9_1_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103737_9_1_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103737_9_1_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103737_9_1_4_Fluorescence.hdr in 11.96399078099057s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103833_10_1_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103833_10_1_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103833_10_1_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103833_10_1_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103833_10_1_3_Fluorescence.hdr in 11.881395780015737s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103930_11_1_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103930_11_1_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103930_11_1_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103930_11_1_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_103930_11_1_2_Fluorescence.hdr in 12.225274713942781s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104026_12_1_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104026_12_1_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104026_12_1_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104026_12_1_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_104026_12_1_1_Fluorescence.hdr in 11.930839544045739s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104122_13_1_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104122_13_1_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104122_13_1_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104122_13_1_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_104122_13_1_0_Fluorescence.hdr in 12.220025382004678s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104219_14_2_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104219_14_2_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104219_14_2_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104219_14_2_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_104219_14_2_0_Fluorescence.hdr in 11.786461286945269s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104306_15_2_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104306_15_2_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104306_15_2_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104306_15_2_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_104306_15_2_1_Fluorescence.hdr in 12.291286980034783s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104403_16_2_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104403_16_2_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104403_16_2_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104403_16_2_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_104403_16_2_2_Fluorescence.hdr in 11.829209820949472s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104459_17_2_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104459_17_2_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104459_17_2_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104459_17_2_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_104459_17_2_3_Fluorescence.hdr in 11.78520084999036s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104556_18_2_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104556_18_2_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104556_18_2_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104556_18_2_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_104556_18_2_4_Fluorescence.hdr in 11.958605900988914s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104652_19_2_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104652_19_2_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104652_19_2_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104652_19_2_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_104652_19_2_5_Fluorescence.hdr in 11.938894344028085s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104748_20_2_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104748_20_2_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104748_20_2_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104748_20_2_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC2_I5.0_F1.9_L100LED30_104748_20_2_6_Fluorescence.hdr in 11.945108084008098s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105108_0_0_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105108_0_0_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105108_0_0_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105108_0_0_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_105108_0_0_0_Fluorescence.hdr in 12.150320793036371s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105150_1_0_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105150_1_0_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105150_1_0_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105150_1_0_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_105150_1_0_1_Fluorescence.hdr in 12.048349858960137s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105337_3_0_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105337_3_0_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105337_3_0_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105337_3_0_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_105337_3_0_3_Fluorescence.hdr in 12.333152661914937s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105433_4_0_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105433_4_0_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105433_4_0_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105433_4_0_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_105433_4_0_4_Fluorescence.hdr in 11.887093435041606s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105529_5_0_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105529_5_0_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105529_5_0_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105529_5_0_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_105529_5_0_5_Fluorescence.hdr in 12.042697565048002s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105626_6_0_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105626_6_0_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105626_6_0_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105626_6_0_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_105626_6_0_6_Fluorescence.hdr in 18.30247234494891s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105722_7_1_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105722_7_1_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105722_7_1_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105722_7_1_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_105722_7_1_6_Fluorescence.hdr in 12.411042826948687s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105810_8_1_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105810_8_1_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105810_8_1_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105810_8_1_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_105810_8_1_5_Fluorescence.hdr in 12.497622874099761s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105906_9_1_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105906_9_1_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105906_9_1_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105906_9_1_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_105906_9_1_4_Fluorescence.hdr in 12.508228449034505s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110003_10_1_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110003_10_1_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110003_10_1_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110003_10_1_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110003_10_1_3_Fluorescence.hdr in 12.11005923198536s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110059_11_1_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110059_11_1_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110059_11_1_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110059_11_1_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110059_11_1_2_Fluorescence.hdr in 12.074751873966306s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110156_12_1_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110156_12_1_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110156_12_1_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110156_12_1_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110156_12_1_1_Fluorescence.hdr in 11.908367536962032s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110252_13_1_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110252_13_1_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110252_13_1_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110252_13_1_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110252_13_1_0_Fluorescence.hdr in 12.116898236097768s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110348_14_2_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110348_14_2_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110348_14_2_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110348_14_2_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110348_14_2_0_Fluorescence.hdr in 11.901763343019411s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110435_15_2_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110435_15_2_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110435_15_2_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110435_15_2_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110435_15_2_1_Fluorescence.hdr in 11.912010372034274s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110532_16_2_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110532_16_2_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110532_16_2_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110532_16_2_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110532_16_2_2_Fluorescence.hdr in 11.967038532020524s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110628_17_2_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110628_17_2_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110628_17_2_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110628_17_2_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110628_17_2_3_Fluorescence.hdr in 12.018726488924585s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110724_18_2_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110724_18_2_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110724_18_2_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110724_18_2_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110724_18_2_4_Fluorescence.hdr in 12.016066761105321s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110821_19_2_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110821_19_2_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110821_19_2_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110821_19_2_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110821_19_2_5_Fluorescence.hdr in 12.169932638993487s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110917_20_2_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110917_20_2_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110917_20_2_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110917_20_2_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_110917_20_2_6_Fluorescence.hdr in 12.247269650921226s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_112529_0_0_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_112529_0_0_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_112529_0_0_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_112529_0_0_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC3_I5.0_F1.9_L100LED30_112529_0_0_2_Fluorescence.hdr in 12.13187712605577s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113039_0_0_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113039_0_0_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113039_0_0_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113039_0_0_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113039_0_0_0_Fluorescence.hdr in 11.883144948980771s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113122_1_0_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113122_1_0_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113122_1_0_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113122_1_0_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113122_1_0_1_Fluorescence.hdr in 11.88064380607102s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113218_2_0_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113218_2_0_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113218_2_0_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113218_2_0_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113218_2_0_2_Fluorescence.hdr in 11.98770192603115s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113315_3_0_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113315_3_0_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113315_3_0_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113315_3_0_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113315_3_0_3_Fluorescence.hdr in 12.043050426989794s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113411_4_0_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113411_4_0_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113411_4_0_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113411_4_0_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113411_4_0_4_Fluorescence.hdr in 11.95687493996229s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113507_5_0_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113507_5_0_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113507_5_0_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113507_5_0_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113507_5_0_5_Fluorescence.hdr in 12.071795561932959s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113604_6_0_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113604_6_0_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113604_6_0_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113604_6_0_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113604_6_0_6_Fluorescence.hdr in 12.251296898000874s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113700_7_1_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113700_7_1_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113700_7_1_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113700_7_1_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113700_7_1_6_Fluorescence.hdr in 11.778604433988221s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113747_8_1_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113747_8_1_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113747_8_1_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113747_8_1_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113747_8_1_5_Fluorescence.hdr in 17.32863012701273s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113844_9_1_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113844_9_1_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113844_9_1_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113844_9_1_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113844_9_1_4_Fluorescence.hdr in 12.225300054065883s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113940_10_1_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113940_10_1_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113940_10_1_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113940_10_1_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_113940_10_1_3_Fluorescence.hdr in 12.222702260012738s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114037_11_1_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114037_11_1_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114037_11_1_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114037_11_1_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114037_11_1_2_Fluorescence.hdr in 12.145504330052063s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114133_12_1_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114133_12_1_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114133_12_1_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114133_12_1_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114133_12_1_1_Fluorescence.hdr in 12.027875721920282s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114229_13_1_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114229_13_1_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114229_13_1_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114229_13_1_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114229_13_1_0_Fluorescence.hdr in 11.993854257976636s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114326_14_2_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114326_14_2_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114326_14_2_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114326_14_2_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114326_14_2_0_Fluorescence.hdr in 11.96694839396514s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114413_15_2_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114413_15_2_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114413_15_2_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114413_15_2_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114413_15_2_1_Fluorescence.hdr in 12.337298346916214s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114509_16_2_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114509_16_2_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114509_16_2_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114509_16_2_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114509_16_2_2_Fluorescence.hdr in 11.891199270961806s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114606_17_2_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114606_17_2_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114606_17_2_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114606_17_2_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114606_17_2_3_Fluorescence.hdr in 12.16030678304378s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114702_18_2_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114702_18_2_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114702_18_2_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114702_18_2_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114702_18_2_4_Fluorescence.hdr in 11.902077648090199s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114759_19_2_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114759_19_2_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114759_19_2_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114759_19_2_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114759_19_2_5_Fluorescence.hdr in 12.201229971949942s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114855_20_2_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114855_20_2_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114855_20_2_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114855_20_2_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC4_I5.0_F1.9_L100LED30_114855_20_2_6_Fluorescence.hdr in 11.855482651968487s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134049_0_0_0_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134049_0_0_0_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134049_0_0_0_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134049_0_0_0_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC5_I5.0_F1.9_L100LED30_134049_0_0_0_Fluorescence.hdr in 11.885095163946971s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134132_1_0_1_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134132_1_0_1_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134132_1_0_1_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134132_1_0_1_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC5_I5.0_F1.9_L100LED30_134132_1_0_1_Fluorescence.hdr in 11.975445827003568s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134228_2_0_2_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134228_2_0_2_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134228_2_0_2_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134228_2_0_2_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC5_I5.0_F1.9_L100LED30_134228_2_0_2_Fluorescence.hdr in 12.209450591006316s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134325_3_0_3_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134325_3_0_3_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134325_3_0_3_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134325_3_0_3_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC5_I5.0_F1.9_L100LED30_134325_3_0_3_Fluorescence.hdr in 12.011688138009049s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134422_4_0_4_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134422_4_0_4_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134422_4_0_4_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134422_4_0_4_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC5_I5.0_F1.9_L100LED30_134422_4_0_4_Fluorescence.hdr in 12.211263752891682s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134518_5_0_5_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134518_5_0_5_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134518_5_0_5_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134518_5_0_5_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC5_I5.0_F1.9_L100LED30_134518_5_0_5_Fluorescence.hdr in 11.986095609958284s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "Running GMOdetector version 0.0.875\n",
      "Saving to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134614_6_0_6_Fluorescence_weights.hdf with key of weights\n",
      "Saving summary stats w/ threshold38.0 to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134614_6_0_6_Fluorescence_summary.csv\n",
      "Saving image to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134614_6_0_6_Fluorescence.png\n",
      "Saving image metadata to /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134614_6_0_6_Fluorescence.csv\n",
      "\n",
      "Finished running sample /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GTENC5_I5.0_F1.9_L100LED30_134614_6_0_6_Fluorescence.hdr in 19.98012329195626s\n",
      "\n",
      "\u001b[?25hR[write to console]: Warning messages:\n",
      "\n",
      "R[write to console]: 1: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n",
      "R[write to console]: 2: \n",
      "R[write to console]: In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  library ‘/usr/lib/R/site-library’ contains no packages\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if [ $parallel -eq 1 ]\n",
    "then\n",
    "    parallel --jobs 20 -a $job_list_name\n",
    "fi\n",
    "\n",
    "if [ $parallel -eq 0 ]\n",
    "then\n",
    "    bash $job_list_name\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Time regression completes:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sun Dec 24 08:51:35 PM PST 2023\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Determine borders of grid in hyperspectral image, which we'll use for alignment and/or cropping to explants later."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Detection of grid edges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selected file: ['chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_Fluorescence.hdr', 'chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_Fluorescence.hdr']\n",
      "Warning: More than one grid file was found. Selecting the one taken last.\n",
      "Selected file: chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_Fluorescence.hdr\n",
      "/usr/local/lib/python3.8/dist-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 130\n",
      "Hyperspectral data loaded, elapsed time: 6.61 seconds\n",
      "Calculations done, elapsed time: 6.62 seconds\n",
      "Coordinates saved, elapsed time: 6.62 seconds\n",
      "find_grid_position.py:66: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "find_grid_position.py:87: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "Plotting done, elapsed time: 7.85 seconds\n",
      "mode,x1,x2,y1,y2\n",
      "hyperspectral,308,1246,138,1297\n",
      "Selected file: ['chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_Fluorescence.hdr', 'chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_Fluorescence.hdr']\n",
      "Warning: More than one grid file was found. Selecting the one taken last.\n",
      "Selected file: chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_Fluorescence.hdr\n",
      "/usr/local/lib/python3.8/dist-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 130\n",
      "Hyperspectral data loaded, elapsed time: 0.04 seconds\n",
      "Calculations done, elapsed time: 0.04 seconds\n",
      "Coordinates saved, elapsed time: 0.05 seconds\n",
      "find_grid_position.py:66: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "find_grid_position.py:87: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "Plotting done, elapsed time: 1.24 seconds\n",
      "mode,x1,x2,y1,y2\n",
      "hyperspectral,138,1297,308,1246\n",
      "Warning: More than one grid file was found. Selecting the one taken last.\n",
      "Selected file: chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_rgb.jpg\n",
      "RGB data loaded, elapsed time: 0.16 seconds\n",
      "Calculations done, elapsed time: 0.18 seconds\n",
      "Coordinates saved, elapsed time: 0.18 seconds\n",
      "find_grid_position.py:66: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "find_grid_position.py:87: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "Plotting done, elapsed time: 1.88 seconds\n",
      "mode,x1,x2,y1,y2\n",
      "RGB,563,3525,1020,3268\n",
      "Warning: More than one grid file was found. Selecting the one taken last.\n",
      "Selected file: chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_rgb.jpg\n",
      "RGB data loaded, elapsed time: 0.14 seconds\n",
      "Calculations done, elapsed time: 0.16 seconds\n",
      "Coordinates saved, elapsed time: 0.16 seconds\n",
      "find_grid_position.py:66: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "find_grid_position.py:87: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "Plotting done, elapsed time: 1.72 seconds\n",
      "mode,x1,x2,y1,y2\n",
      "RGB,1020,3268,563,3525\n",
      "Hyperspectral, original orientation:      308,138,1246,1297\n",
      "Hyperspectral, transf. for labels on bot: 138,308,1297,1246\n",
      "RGB, rotated and scaled for Densenet:     282,510,1762,1634\n",
      "RGB, oriented for alignment with hyp:     1020,563,3268,3525\n"
     ]
    }
   ],
   "source": [
    "# Function to process orientation and extract grid borders\n",
    "process_orientation() {\n",
    "    local mode=$1\n",
    "    local rotation=$2\n",
    "    local flip_horizontal=$3\n",
    "    local label=$4\n",
    "    local half_scale=$5\n",
    "\n",
    "    # Navigate to the intermediates directory and run the Python script\n",
    "    cd \"${cwd}/intermediates/\"\n",
    "    command=\"python3.8 find_grid_position.py --mode $mode --cap 500 --index 130 --data $data --plot --rotation $rotation\"\n",
    "    #command=\"python3.8 ${cwd}/intermediates/find_grid_position.py --mode $mode --cap 500 --index 130 --data $data --plot --rotation $rotation\"\n",
    "    if [ \"$flip_horizontal\" = true ]; then\n",
    "        command=\"$command --flip_horizontal\"\n",
    "    fi\n",
    "    eval $command\n",
    "\n",
    "    # Navigate to the data directory where coordinates.csv is saved\n",
    "    cd \"$data\"\n",
    "    cat coordinates.csv\n",
    "\n",
    "    # Read the CSV file and extract x1, x2, y1, y2\n",
    "    firstline=1\n",
    "    while IFS=',' read -r mode x1 x2 y1 y2; do\n",
    "        if [ \"$firstline\" -eq \"0\" ]; then\n",
    "            if [ \"$half_scale\" = true ]; then\n",
    "                # Divide by 2 and round to integer\n",
    "                x1=$(printf \"%.0f\" $(echo \"$x1 / 2\" | bc -l))\n",
    "                x2=$(printf \"%.0f\" $(echo \"$x2 / 2\" | bc -l))\n",
    "                y1=$(printf \"%.0f\" $(echo \"$y1 / 2\" | bc -l))\n",
    "                y2=$(printf \"%.0f\" $(echo \"$y2 / 2\" | bc -l))\n",
    "            fi\n",
    "            eval \"$label=\\\"$x1,$y1,$x2,$y2\\\"\"\n",
    "        fi\n",
    "        firstline=0\n",
    "    done < coordinates.csv\n",
    "}\n",
    "\n",
    "# Process original orientation\n",
    "process_orientation \"hyperspectral\" 0 false \"aligned_grid_borders_original\" false\n",
    "\n",
    "# Process label bottom orientation\n",
    "process_orientation \"hyperspectral\" 270 true \"aligned_grid_borders_label_bottom\" false\n",
    "\n",
    "# Process for DenseNet model orientation\n",
    "process_orientation \"RGB\" 270 false \"pre_aligned_resized_grid_borders_densenet\" true\n",
    "\n",
    "# Process for hyperspectral layer matching orientation\n",
    "process_orientation \"RGB\" 180 true \"pre_aligned_grid_borders_hyp_oriented\" false\n",
    "\n",
    "\n",
    "# Print results\n",
    "echo \"Hyperspectral, original orientation:      $aligned_grid_borders_original\"\n",
    "echo \"Hyperspectral, transf. for labels on bot: $aligned_grid_borders_label_bottom\"\n",
    "echo \"RGB, rotated and scaled for Densenet:     $pre_aligned_resized_grid_borders_densenet\"\n",
    "echo \"RGB, oriented for alignment with hyp:     $pre_aligned_grid_borders_hyp_oriented\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Semantic segmentation of tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Images are segmented into specific plant tissues by a deep neural network of the state-of-the-art Deeplab v3 architecture <a href=\"https://arxiv.org/abs/1706.05587\" target=\"_blank\">Liang-Chieh et al., 2017</a>. The model has been trained using training sets generated with our annotation GUI Intelligent DEep Annotator for Segmentation (IDEAS, available on <a href=\"https://bitbucket.org/JialinYuan/image-annotator/src/master/\" target=\"_blank\">Bitbucket</a>, publication pending). Our branch of the Deeplab v3 repo, including a Jupyter walkthrough for training, can be found on Github."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Training is completed upstream of this notebook, which only entails analysis of test data using the latest model."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/downsized/segmentation_composite2.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure: This example image was taken from an experiment on the effects of different CIMs on cottonwood regeneration. This composite image illustrates that for every sample, tissues are segmented into stem (red), callus (blue) and shoot (green). These composite images, useful for manual inspection of results, are produced when the 'composite' option is on."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pre-processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Normalize orientation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We desire for images to all be in the same orientation. At one point, the camera on the *macroPhor Array* was set to automatically detect orientation, which led to images randomly being in portrait or landscape. Here we will standardize the orientation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for filename in $data/*.jpg; do\n",
    "    exiftool -Orientation=8 -n $filename > ${data}log_exiftool.txt\n",
    "    done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f $data/*original*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Crop and resize"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This script resizes images to 900x900 and then crops away top and bottom 150 pixels for a final image size of 900x600."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The purpose for cropping is to remove labels, which has been standard practice for all training and testing. Otherwise, we could run into problems such as the neural network \"learning\" plants labeled as control have more or less regeneration.<br>The purpose for resizing is to reduce computational expense."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "if [ \"$segmentation_mode\" = \"rgb\" ]; then\n",
    "    cd ${cwd}/intermediates/\n",
    "    python crop.py $data\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Prepare input list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The script `inference.py` requires a list of all files to be analyzed. We will create this file as `test.csv`. This will be a list of all our (pre-processed) image files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "if [ \"$segmentation_mode\" = \"rgb\" ]; then\n",
    "    cd $data\n",
    "    ls -d $PWD/* $data | grep -i \"rgb_cropped.jpg\" > test.csv\n",
    "    sed -i '/hroma/d' \"${data}/test.csv\"\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Inference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The trained model is deployed to perform semantic segmentation of experimental images. A list of RGB images to be segmented by the trained model is passed through the --image-list option. For each of these images, we will obtain an output mask (.png) of labeled tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `opencv`, `scipy`, `yaml` and `tensorflow` (version 1.14)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:From deeplab/inference.py:142: The name tf.app.run is deprecated. Please use tf.compat.v1.app.run instead.\n",
      "\n",
      "WARNING:tensorflow:From deeplab/inference.py:51: The name tf.gfile.GFile is deprecated. Please use tf.io.gfile.GFile instead.\n",
      "\n",
      "W1224 20:52:38.043788 140365967024640 deprecation_wrapper.py:119] From deeplab/inference.py:51: The name tf.gfile.GFile is deprecated. Please use tf.io.gfile.GFile instead.\n",
      "\n",
      "WARNING:tensorflow:From deeplab/inference.py:52: The name tf.GraphDef is deprecated. Please use tf.compat.v1.GraphDef instead.\n",
      "\n",
      "W1224 20:52:38.043966 140365967024640 deprecation_wrapper.py:119] From deeplab/inference.py:52: The name tf.GraphDef is deprecated. Please use tf.compat.v1.GraphDef instead.\n",
      "\n",
      "WARNING:tensorflow:From deeplab/inference.py:61: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.\n",
      "\n",
      "W1224 20:52:38.925038 140365967024640 deprecation_wrapper.py:119] From deeplab/inference.py:61: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.\n",
      "\n",
      "2023-12-24 20:52:38.925377: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA\n",
      "2023-12-24 20:52:38.942203: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3499910000 Hz\n",
      "2023-12-24 20:52:38.944381: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x25d9dc0 executing computations on platform Host. Devices:\n",
      "2023-12-24 20:52:38.944437: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2023-12-24 20:52:43.016252: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "WARNING:tensorflow:From /mnt/models/rgb/poplar_model_2_w_contam/deeplab/utils/save_annotation.py:82: The name tf.gfile.Open is deprecated. Please use tf.io.gfile.GFile instead.\n",
      "\n",
      "W1224 20:52:44.964853 140365967024640 deprecation_wrapper.py:119] From /mnt/models/rgb/poplar_model_2_w_contam/deeplab/utils/save_annotation.py:82: The name tf.gfile.Open is deprecated. Please use tf.io.gfile.GFile instead.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if [ \"$segmentation_mode\" = \"rgb\" ]; then\n",
    "    export PYTHONPATH=$PYTHONPATH:`pwd`:`pwd`/slim\n",
    "    cd $deeplab_path\n",
    "    python3.7 -W ignore deeplab/inference.py \\\n",
    "        --image_lists \"${data}/test.csv\" \\\n",
    "        --crop_size 900 --crop_size 600 \\\n",
    "        --seg_results segmentation_results \\\n",
    "        --model_dir \"${deeplab_path}/deeplab/model/\" \\\n",
    "        >> $data/log_inference.txt\n",
    "    mv \"${deeplab_path}/segmentation_results/raw/\"* $data/\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if [ \"$segmentation_mode\" = \"hyperspectral\" ]; then\n",
    "    cd $cubeml_path\n",
    "    #cd /mnt/cubeml/\n",
    "    python scripts/batch_inference.py \\\n",
    "    --dir $data \\\n",
    "    --pickle $segmentation_model_path \\\n",
    "    --method $segmentation_model_type \\\n",
    "    --false_color\n",
    "    >> $data/log_inference.txt\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Post-processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Name outputs to reflect that they are segmentation results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "if [ \"$segmentation_mode\" = \"rgb\" ]; then\n",
    "    cd $data\n",
    "    for file in *_rgb_cropped.png; do mv -f \"$file\" \"${file%_rgb_cropped.png}_segment_cropped.png\"; done\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Re-expand segment outputs to same size as original RGB files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "if [ \"$segmentation_mode\" = \"rgb\" ]; then\n",
    "    cd $alignment_path\n",
    "    python expand.py $data >> $data/log_expand.txt\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make composite images with side-by-side RGB, segmentation outputs and blended images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "if [ $composite -eq 1 ]\n",
    "then\n",
    "    echo \"making composites\"\n",
    "    cd $gmolabeler_path\n",
    "    python image_blender.py $data 0.75 'both' 1 0\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Classification of contaminated/missing explants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Plates are cropped into sub-images for each explant and each is analyzed to determine if the explant position should be excluded from analysis due to being missing or contamination. Missing and contaminated explants are recognized using a trained Densenet model (<a href=\"https://github.com/Contamination-Classification/DenseNet\" target=\"_blank\">Huang, et al. 2018</a>). Our fork of the Densenet repository is available on <a href=\"https://arxiv.org/abs/1608.06993\" target=\"_blank\">GitHub</a>."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/Densenet.png\">\n",
    "Figure: These are four examples of contaminated explants used in the training set for this pre-trained model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To check the grid cropping dimensions, we can run the following script. Note that these are the dimensions to crop the image to after resizing to 2000x2000 (from 4000x4000 in the case of the *macroPhor Array*)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prepare list of images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing explants will be inferred.\n"
     ]
    }
   ],
   "source": [
    "if [ $missing_explants = \"Automatic\" ]; then\n",
    "   echo \"Missing explants will be inferred.\"\n",
    "   cd $data\n",
    "   ls -d $PWD/* $data | grep -i \"rgb.jpg\" > rgb_list.txt\n",
    "   sed -i '/hroma/d' rgb_list.txt\n",
    "   img_list_path=\"${data}/rgb_list.txt\"\n",
    "else\n",
    "   echo \"Missing explants input manually by user, in file: \"\n",
    "   echo $missing_explants\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If the mode for missing explant data is automatic, prepare input file for script to detect missing explants and run this script."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Infer contaminated/missing explants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using TensorFlow backend.\n",
      "2023-12-24 20:55:48.693664: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA\n",
      "2023-12-24 20:55:48.698068: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3499910000 Hz\n",
      "2023-12-24 20:55:48.700043: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x2d38c90 executing computations on platform Host. Devices:\n",
      "2023-12-24 20:55:48.700097: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2023-12-24 20:55:48.710183: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "WARNING:tensorflow:From /home/gmobot/.local/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:4070: The name tf.nn.max_pool is deprecated. Please use tf.nn.max_pool2d instead.\n",
      "\n",
      "WARNING:tensorflow:From /home/gmobot/.local/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:4074: The name tf.nn.avg_pool is deprecated. Please use tf.nn.avg_pool2d instead.\n",
      "\n",
      "WARNING:tensorflow:From /home/gmobot/.local/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:422: The name tf.global_variables is deprecated. Please use tf.compat.v1.global_variables instead.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if [ $missing_explants = \"Automatic\" ]; then\n",
    "    cd $data\n",
    "    python3.7 -W ignore ${contamination_path}/inference.py \\\n",
    "    --img-list=$img_list_path \\\n",
    "    --crop_dims $pre_aligned_resized_grid_borders_densenet \\\n",
    "    --weights_path $densenet_model_path \\\n",
    "    --output_file=\"${data}/output.csv\" >> $data/log_contam.txt\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing explants inferred by model and written to file:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//output.csv\n"
     ]
    }
   ],
   "source": [
    "if [ $missing_explants = \"Automatic\" ]; then\n",
    "    missing_explants=\"${data}/output.csv\"\n",
    "    echo \"Missing explants inferred by model and written to file:\"\n",
    "    echo $missing_explants\n",
    "else\n",
    "    echo \"Missing explants input manually by user, in file: \"\n",
    "    echo $missing_explants\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Alignment of RGB and hyperspectral layers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To match the frame and angle of RGB and hyperspectral image layers, we perform a homography transformation using a method described [in these notebooks](https://github.com/naglemi/GMOnotebook/tree/master/1_Decide_parameters/2_Align_and_crop_parameters/2_find_alignment_parameters). Using a pair of standard images, a homography matrix is calculated for the necessary transformation of RGB images to align with hyperspectral images. The transformation can then be applied to large batches of images rapidly, as long as the RGB and hyperspectral cameras remain in the same positions relative to one another (as they do in the macroPhor Array platform)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/Alignment.png\">\n",
    "Figure: To enable precise calculation of a homography matrix for transformation of RGB images to match hyperspectral images, we used images of a piece of paper with grid marks."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original Borders: 308,138,1246,1297\n",
      "Label Bottom Borders: 138,308,1297,1246\n",
      "DenseNet Borders: 282,510,1762,1634\n",
      "Hyperspectral Oriented Borders: 1020,563,3268,3525\n"
     ]
    }
   ],
   "source": [
    "echo \"Original Borders: $aligned_grid_borders_original\"\n",
    "echo \"Label Bottom Borders: $aligned_grid_borders_label_bottom\"\n",
    "echo \"DenseNet Borders: $pre_aligned_resized_grid_borders_densenet\"\n",
    "echo \"Hyperspectral Oriented Borders: $pre_aligned_grid_borders_hyp_oriented\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of images:\n",
      "184\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100838_1_0_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_100838_1_0_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100838_1_0_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100934_2_0_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_100934_2_0_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_100934_2_0_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101031_3_0_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101031_3_0_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101031_3_0_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101127_4_0_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101127_4_0_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101127_4_0_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101223_5_0_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101223_5_0_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101223_5_0_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101320_6_0_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101320_6_0_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101320_6_0_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101416_7_1_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101416_7_1_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101416_7_1_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101503_8_1_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1571)\n",
      "Shape rgb:\n",
      "(1419, 1571)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101503_8_1_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101503_8_1_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1571)\n",
      "Shape rgb:\n",
      "(1419, 1571)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101600_9_1_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101600_9_1_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101600_9_1_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101656_10_1_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101656_10_1_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101656_10_1_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101753_11_1_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101753_11_1_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101753_11_1_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101849_12_1_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101849_12_1_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101849_12_1_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101946_13_1_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_101946_13_1_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_101946_13_1_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102042_14_2_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_102042_14_2_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102042_14_2_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102129_15_2_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_102129_15_2_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102129_15_2_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102225_16_2_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_102225_16_2_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102225_16_2_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102322_17_2_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1572)\n",
      "Shape rgb:\n",
      "(1419, 1572)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_102322_17_2_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102322_17_2_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1572)\n",
      "Shape rgb:\n",
      "(1419, 1572)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102418_18_2_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_102418_18_2_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102418_18_2_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102515_19_2_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_102515_19_2_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102515_19_2_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102611_20_2_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC1_I5.0_F1.9_L100LED30_102611_20_2_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC1_I5.0_F1.9_L100LED30_102611_20_2_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_102933_0_0_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_102933_0_0_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_102933_0_0_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103015_1_0_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103015_1_0_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103015_1_0_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103112_2_0_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103112_2_0_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103112_2_0_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103208_3_0_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103208_3_0_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103208_3_0_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103305_4_0_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103305_4_0_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103305_4_0_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103401_5_0_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103401_5_0_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103401_5_0_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103457_6_0_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103457_6_0_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103457_6_0_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103554_7_1_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103554_7_1_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103554_7_1_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103641_8_1_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103641_8_1_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103641_8_1_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103737_9_1_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103737_9_1_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103737_9_1_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103833_10_1_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103833_10_1_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103833_10_1_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103930_11_1_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_103930_11_1_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_103930_11_1_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104026_12_1_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_104026_12_1_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104026_12_1_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104122_13_1_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_104122_13_1_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104122_13_1_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104219_14_2_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_104219_14_2_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104219_14_2_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104306_15_2_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_104306_15_2_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104306_15_2_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104403_16_2_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_104403_16_2_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104403_16_2_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104459_17_2_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_104459_17_2_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104459_17_2_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104556_18_2_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_104556_18_2_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104556_18_2_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104652_19_2_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_104652_19_2_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104652_19_2_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104748_20_2_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC2_I5.0_F1.9_L100LED30_104748_20_2_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC2_I5.0_F1.9_L100LED30_104748_20_2_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105108_0_0_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_105108_0_0_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105108_0_0_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105150_1_0_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_105150_1_0_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105150_1_0_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105337_3_0_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_105337_3_0_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105337_3_0_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105433_4_0_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_105433_4_0_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105433_4_0_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105529_5_0_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_105529_5_0_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105529_5_0_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105626_6_0_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_105626_6_0_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105626_6_0_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105722_7_1_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_105722_7_1_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105722_7_1_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105810_8_1_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_105810_8_1_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105810_8_1_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105906_9_1_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_105906_9_1_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_105906_9_1_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110003_10_1_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110003_10_1_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110003_10_1_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110059_11_1_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110059_11_1_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110059_11_1_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110156_12_1_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110156_12_1_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110156_12_1_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110252_13_1_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110252_13_1_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110252_13_1_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110348_14_2_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110348_14_2_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110348_14_2_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110435_15_2_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110435_15_2_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110435_15_2_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110532_16_2_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110532_16_2_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110532_16_2_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110628_17_2_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110628_17_2_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110628_17_2_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110724_18_2_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110724_18_2_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110724_18_2_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110821_19_2_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110821_19_2_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110821_19_2_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110917_20_2_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_110917_20_2_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_110917_20_2_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_112529_0_0_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1571)\n",
      "Shape rgb:\n",
      "(1419, 1571)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC3_I5.0_F1.9_L100LED30_112529_0_0_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC3_I5.0_F1.9_L100LED30_112529_0_0_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1571)\n",
      "Shape rgb:\n",
      "(1419, 1571)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113039_0_0_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113039_0_0_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113039_0_0_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113122_1_0_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113122_1_0_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113122_1_0_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113218_2_0_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113218_2_0_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113218_2_0_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113315_3_0_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113315_3_0_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113315_3_0_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113411_4_0_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113411_4_0_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113411_4_0_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113507_5_0_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113507_5_0_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113507_5_0_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113604_6_0_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113604_6_0_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113604_6_0_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113700_7_1_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113700_7_1_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113700_7_1_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113747_8_1_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113747_8_1_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113747_8_1_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113844_9_1_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113844_9_1_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113844_9_1_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113940_10_1_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_113940_10_1_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_113940_10_1_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114037_11_1_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114037_11_1_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114037_11_1_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114133_12_1_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114133_12_1_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114133_12_1_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114229_13_1_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114229_13_1_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114229_13_1_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1566)\n",
      "Shape rgb:\n",
      "(1419, 1566)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114326_14_2_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114326_14_2_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114326_14_2_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114413_15_2_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114413_15_2_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114413_15_2_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114509_16_2_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114509_16_2_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114509_16_2_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114606_17_2_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114606_17_2_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114606_17_2_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114702_18_2_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114702_18_2_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114702_18_2_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114759_19_2_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114759_19_2_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114759_19_2_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114855_20_2_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC4_I5.0_F1.9_L100LED30_114855_20_2_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC4_I5.0_F1.9_L100LED30_114855_20_2_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134049_0_0_0_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC5_I5.0_F1.9_L100LED30_134049_0_0_0_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134049_0_0_0_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134132_1_0_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC5_I5.0_F1.9_L100LED30_134132_1_0_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134132_1_0_1_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1570)\n",
      "Shape rgb:\n",
      "(1419, 1570)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134228_2_0_2_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC5_I5.0_F1.9_L100LED30_134228_2_0_2_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134228_2_0_2_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134325_3_0_3_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC5_I5.0_F1.9_L100LED30_134325_3_0_3_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134325_3_0_3_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1568)\n",
      "Shape rgb:\n",
      "(1419, 1568)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134422_4_0_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC5_I5.0_F1.9_L100LED30_134422_4_0_4_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134422_4_0_4_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134518_5_0_5_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC5_I5.0_F1.9_L100LED30_134518_5_0_5_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134518_5_0_5_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1569)\n",
      "Shape rgb:\n",
      "(1419, 1569)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134614_6_0_6_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1573)\n",
      "Shape rgb:\n",
      "(1419, 1573)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/GTENC5_I5.0_F1.9_L100LED30_134614_6_0_6_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/GTENC5_I5.0_F1.9_L100LED30_134614_6_0_6_segment_uncropped.png\n",
      "rotating mask\n",
      "img mode before conversion:\n",
      "P\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1573)\n",
      "Shape rgb:\n",
      "(1419, 1573)\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_overlay.jpg\n",
      "Performing alignment for image:\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_rgb.jpg\n",
      "img mode before conversion:\n",
      "RGB\n",
      "img mode after conversion:\n",
      "RGB\n",
      "Will apply matrix:\n",
      "<ProjectiveTransform(matrix=\n",
      "    [[ 4.17259786e-01, -1.97488717e-16, -1.17604982e+02],\n",
      "     [ 3.47114381e-16,  3.91289669e-01, -8.22960837e+01],\n",
      "     [ 2.06238847e-19, -1.28696716e-19,  1.00000000e+00]])>\n",
      "Shape hyp:\n",
      "(1419, 1567)\n",
      "Shape rgb:\n",
      "(1419, 1567)\n",
      "Saving overlay to:\n",
      "/mnt/output/debug/chromagrid_I5.0_F1.9_L100LED30_094541_1_1_4_overlay.jpg\n"
     ]
    }
   ],
   "source": [
    "# Assuming that the borders are provided as four integers each, in the order: left, top, right, bottom\n",
    "pre_aligned_borders=(${pre_aligned_grid_borders_hyp_oriented//,/ })\n",
    "aligned_borders=(${aligned_grid_borders_original//,/ })\n",
    "\n",
    "# Call the new script with these borders and other necessary arguments\n",
    "if [ \"$segmentation_mode\" = \"rgb\" ]; then\n",
    "    python3.8 ${cwd}/intermediates/batch_align_auto.py \\\n",
    "    --left_source ${pre_aligned_borders[0]} \\\n",
    "    --top_source ${pre_aligned_borders[1]} \\\n",
    "    --right_source ${pre_aligned_borders[2]} \\\n",
    "    --bottom_source ${pre_aligned_borders[3]} \\\n",
    "    --left_target ${aligned_borders[0]} \\\n",
    "    --top_target ${aligned_borders[1]} \\\n",
    "    --right_target ${aligned_borders[2]} \\\n",
    "    --bottom_target ${aligned_borders[3]} \\\n",
    "    --img_dir $data \\\n",
    "    --hyp_path $output_directory_full \\\n",
    "    --channel_index 130 \\\n",
    "    --overlay_dir \"/mnt/output/debug/\"\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "if [ \"$segmentation_mode\" = \"rgb\" ]; then\n",
    "    hypercube_jpg=$(echo $hypercube_csv | sed -e 's/\\.csv/.jpg/g')\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cross-analyze deep segmentation and regression results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Scripts in the <a href=\"https://github.com/naglemi/GMOlabeler\" target=\"_blank\">GMOlabeler repository</a> are used to cross-reference results from deep segmentation of RGB images and regression of hyperspectral imaging, apply thresholding parameters to classify tissues as transgenic or escapes, and produce plots."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/GMOlabeler.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure: The various steps of data processing in GMOlabeler are illustrated for an example explant from an experiment on CIM optimization for cottonwood. Images of plates are cropped to a sub-image for each explant. RGB segmentation results and hyperspectral regression results are cross-referenced to calculate fluorecent proteins in specific tissues and infer whether these tissues are transgenic."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prepare sample datasheet input"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Prepare input file we will use for making plots. This file contains paths to CLS results, RGB images, and hyperspectral images."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "”2023-12-24”\n"
     ]
    }
   ],
   "source": [
    "echo $datestamp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h[1] \"”2023-12-24”\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h[1] \"Looking for CLS data in: /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//\"\n",
      "\u001b[?25h[1] \"2023-12-24\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h[1] \"How many RGB files? 93\"\n",
      "\u001b[?25h\u001b[?25h[1] \"Looking for CLS files in directory /mnt/output/gmodetector_out/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h[1] \"How many CLS files? 93\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h[1] \"Writing 91 rows to /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///samples_pre_labeling.csv\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\n"
     ]
    }
   ],
   "source": [
    "cd \"${cwd}/intermediates/\"\n",
    "\n",
    "# Check if segmentation_mode is set to \"hyperspectral\"\n",
    "if [ \"$segmentation_mode\" = \"hyperspectral\" ]; then\n",
    "  Rscript pre_label.R \\\n",
    "  -r \"${data}/\" \\\n",
    "  -R \"${output_directory_prefix}\" \\\n",
    "  -i 1 \\\n",
    "  -d $datestamp \\\n",
    "  --segmentation_model_key $segmentation_model_key # Only included if segmentation_mode is hyperspectral\n",
    "else\n",
    "  Rscript pre_label.R \\\n",
    "  -r \"${data}/\" \\\n",
    "  -R \"${output_directory_prefix}\" \\\n",
    "  -i 1 \\\n",
    "  -d $datestamp\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No grid found at a path given by user, searching for one...\n",
      "Grid found: /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_rgb_processed.png\n"
     ]
    }
   ],
   "source": [
    "# Check if $aligned_grid is unset or the file doesn't exist\n",
    "if [[ -z \"$aligned_grid\" || ! -f \"$aligned_grid\" ]]; then\n",
    "  echo \"No grid found at a path given by user, searching for one...\"\n",
    "\n",
    "  # Find files, extract filenames and timestamps, sort by timestamp, and get the filename with the second-to-last timestamp\n",
    "  #  in Strauss Lab, this will always represent the grid with bold numbers and lines, useful for visualizing grid cropping\n",
    "  aligned_grid=$(find \"$data\" -maxdepth 1 -type f -name \"*hroma*rgb_processed.png*\" ! -name \"*csv*\" \\\n",
    "                 | while read -r file; do\n",
    "                     timestamp=$(echo \"$file\" | grep -o '[0-9]\\{6\\}')\n",
    "                     echo \"$timestamp $file\"\n",
    "                   done \\\n",
    "                 | sort -k1,1nr \\\n",
    "                 | head -n 2 \\\n",
    "                 | tail -n 1 \\\n",
    "                 | cut -d' ' -f2-)\n",
    "\n",
    "  # Check if a file was found\n",
    "  if [[ -z \"$aligned_grid\" ]]; then\n",
    "    echo \"No suitable grid file found.\"\n",
    "  else\n",
    "    # Output the found path\n",
    "    echo \"Grid found: $aligned_grid\"\n",
    "  fi\n",
    "fi\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "138,308,1297,1246\n"
     ]
    }
   ],
   "source": [
    "echo $aligned_grid_borders_label_bottom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//samples_pre_labeling.csv\n",
      "/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9/chromagrid_I5.0_F1.9_L100LED30_094450_0_0_1_rgb_processed.png\n",
      "38\n",
      "Chl\n",
      "12\n",
      "hdf\n",
      "/mnt/output/gmodetector_out//gmolabeler_logic_outputs/\n"
     ]
    }
   ],
   "source": [
    "echo ${data}/samples_pre_labeling.csv\n",
    "echo $aligned_grid\n",
    "echo $reporter_threshold\n",
    "echo \"Chl\"\n",
    "echo $grid\n",
    "echo \"hdf\"\n",
    "echo ${output_directory_prefix}/gmolabeler_logic_outputs/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "aligned_grid_borders_label_bottom_formatted=$(echo $aligned_grid_borders_original | awk -F',' '{print $2, $4, $3, $1}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "138 1297 1246 308\n"
     ]
    }
   ],
   "source": [
    "echo $aligned_grid_borders_label_bottom_formatted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "aligned_grid_borders_label_bottom_formatted=$(echo $aligned_grid_borders_original | awk -F',' '{print $2, $4, $3, $1}')\n",
    "cd $gmolabeler_path\n",
    "for reporter in ${reporters[@]}; do\n",
    "    # Start the command with the basic arguments\n",
    "    cmd=\"python main.py \\\n",
    "    \\\"${data}/samples_pre_labeling.csv\\\" \\\n",
    "    $aligned_grid \\\n",
    "    $reporter_threshold \\\n",
    "    $reporter \\\n",
    "    $grid \\\n",
    "    \\\"hdf\\\" \\\n",
    "    \\\"${output_directory_prefix}/gmolabeler_logic_outputs/\\\"\"\n",
    "\n",
    "    # Append the grid borders to the command\n",
    "    if [[ -n \"$aligned_grid_borders_label_bottom_formatted\" ]]; then\n",
    "        cmd+=\" \\\"$aligned_grid_borders_label_bottom_formatted\\\"\"\n",
    "    fi\n",
    "\n",
    "    # Check if segmentation_model_key is set and points to a file\n",
    "    if [[ -n $segmentation_model_key && -f $segmentation_model_key ]]; then\n",
    "        # Append the segmentation model key to the command as a named argument\n",
    "        cmd+=\" --segmentation_model_key \\\"$segmentation_model_key\\\"\"\n",
    "    fi\n",
    "\n",
    "    # Run the command and redirect stdout to the log file\n",
    "    eval $cmd > \"$data/log_gmolabeler_$reporter.txt\"\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Calculate sums of statistics over combined segments"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are interested in all regenerated tissue (callus + shoot) as well as all tissue (including stem as well). We will calculate aggregate statistics over these groups."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "echo $segmentation_model_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Rscript calculate_sum_stats_over_combined_segments.R --output_dir \"/mnt/output/gmodetector_out//gmolabeler_logic_outputs/\" --datapath \"Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GFP/\" --exclude_tissues \"Background Stem Necrotic Explant\"\n",
      "[1] \"Completed for whole-tissue batch\"\n",
      "[1] \"Now to compute only for regenerated tissues\"\n",
      "[1] \"Rows before excluding: \"\n",
      "[1] 4369\n",
      "[1] \"Rows after excluding: \"\n",
      "[1] 4369\n",
      "[1] \"Writing to /mnt/output/gmodetector_out//gmolabeler_logic_outputs/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GFP/stats_with_sums_over_tissues.csv\"\n",
      "[1] \"Script completed successfully.\"\n",
      "\n",
      "Rscript calculate_sum_stats_over_combined_segments.R --output_dir \"/mnt/output/gmodetector_out//gmolabeler_logic_outputs/\" --datapath \"Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//Chl/\" --exclude_tissues \"Background Stem Necrotic Explant\"\n",
      "[1] \"Completed for whole-tissue batch\"\n",
      "[1] \"Now to compute only for regenerated tissues\"\n",
      "[1] \"Rows before excluding: \"\n",
      "[1] 4369\n",
      "[1] \"Rows after excluding: \"\n",
      "[1] 4369\n",
      "[1] \"Writing to /mnt/output/gmodetector_out//gmolabeler_logic_outputs/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//Chl/stats_with_sums_over_tissues.csv\"\n",
      "[1] \"Script completed successfully.\"\n"
     ]
    }
   ],
   "source": [
    "cd $gmolabeler_path\n",
    "for reporter in \"${reporters[@]}\"; do\n",
    "    # Start the command with the base part\n",
    "    cmd=\"Rscript calculate_sum_stats_over_combined_segments.R \\\n",
    "    --output_dir \\\"${output_directory_prefix}/gmolabeler_logic_outputs/\\\" \\\n",
    "    --datapath \\\"${data_folder}/${reporter}/\\\"\"\n",
    "\n",
    "    # Append the model key path if it's set and not None\n",
    "    if [ -n \"${segmentation_model_key}\" ] && [ \"${segmentation_model_key}\" != \"None\" ]; then\n",
    "        cmd+=\" --keypath \\\"${segmentation_model_key}\\\"\"\n",
    "    fi\n",
    "\n",
    "    # Append the exclude tissues string if it's set and not None\n",
    "    if [ -n \"${unregenerated_tissues}\" ] && [ \"${unregenerated_tissues}\" != \"None\" ]; then\n",
    "        echo $exclude_tissues\n",
    "        cmd+=\" --exclude_tissues \\\"${unregenerated_tissues}\\\"\"\n",
    "    fi\n",
    "    \n",
    "    echo $cmd\n",
    "\n",
    "    # Execute the command\n",
    "    eval $cmd\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make plots of results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rscript grid_item_plots.R -d \"Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//\" -r \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/GTNEC_labels.xlsx\" -p 3 -v categorical -m 1 -M /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//output.csv -g 12 --samples-pre-labeling /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//samples_pre_labeling.csv --sort 1 --height 5 --width 9 --Reporter GFP --outdir \"/mnt/output/gmodetector_out/\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25hWarning message:\n",
      "In storage.mode(default) <- type : NAs introduced by coercion\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h$datapath1\n",
      "[1] \"Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//\"\n",
      "\n",
      "$randomization_datasheet_path\n",
      "[1] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/GTNEC_labels.xlsx\"\n",
      "\n",
      "$`samples-pre-labeling`\n",
      "[1] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//samples_pre_labeling.csv\"\n",
      "\n",
      "$pixel_threshold\n",
      "[1] 3\n",
      "\n",
      "$variable\n",
      "[1] \"categorical\"\n",
      "\n",
      "$missing\n",
      "[1] 1\n",
      "\n",
      "$MissingList\n",
      "[1] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//output.csv\"\n",
      "\n",
      "$grid_type\n",
      "[1] 12\n",
      "\n",
      "$sort\n",
      "[1] 1\n",
      "\n",
      "$height\n",
      "[1] 5\n",
      "\n",
      "$width\n",
      "[1] 9\n",
      "\n",
      "$debug\n",
      "[1] 1\n",
      "\n",
      "$Reporter\n",
      "[1] \"GFP\"\n",
      "\n",
      "$keypath\n",
      "[1] \"FALSE\"\n",
      "\n",
      "$outdir\n",
      "[1] \"/mnt/output/gmodetector_out/\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h[1] \"Saving list of input arguments to : /mnt/output/gmodetector_out//gmolabeler_stats_plots/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GFP/gridplot_args.rds\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h[1] \"Reading in output from GMOlabeler at path: /mnt/output/gmodetector_out//gmolabeler_logic_outputs/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//GFP/stats_with_sums_over_tissues.csv\"\n",
      "\u001b[?25h[1] \"Assuming default tissues for original poplar model because not specified with `keypath`\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"Rows in output from GMOlabeler: 6552\"\n",
      "\n",
      "[1] \"Max n_pixels_passing_threshold in output from GMOlabeler: 1428\"\n",
      "\n",
      "[1] \"Max total_signal in output from GMOlabeler: 583473.316290787\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"Look at the top of output from GMOlabeler\"\n",
      "[1] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[2] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[3] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[4] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[5] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[6] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "\n",
      "[1] \"Look at the top of missing explant data\"\n",
      "\u001b[38;5;246m# A tibble: 0 × 3\u001b[39m\n",
      "\u001b[38;5;246m# ℹ 3 variables: filename <chr>, grid_item <chr>, present <chr>\u001b[39m\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"Look at the head of filenames for missing explant data\"\n",
      "character(0)\n",
      "\n",
      "[1] \"Look at the colnames of output from GMOlabeler\"\n",
      " [1] \"V1\"                         \"filename\"                  \n",
      " [3] \"grid_item\"                  \"segment_hex\"               \n",
      " [5] \"intensity_threshold\"        \"n_pixels_passing_threshold\"\n",
      " [7] \"mean_signal\"                \"max_signal\"                \n",
      " [9] \"total_signal\"               \"total_pixels\"              \n",
      "\n",
      "[1] \"Look at the colnames of missing explant data\"\n",
      "[1] \"filename\"  \"grid_item\" \"present\"  \n",
      "\n",
      "[1] \"Before merging the GMOlabeler and missing output data,they have rows of 6552 and 0 respectively.\"\n",
      "\u001b[?25h\u001b[?25h\n",
      "[1] \"After merging but before removing missing/contminated explants, merged output has 6552 rows.\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"After removing the missing/contaminated explants, merged output has 6552\"\n",
      "\u001b[?25h\n",
      "[1] \"After processing GMOlabeler output to merge with missing explantdata, Head of data (w/ first 5 col): \"\n",
      "                                      filename grid_item          V1\n",
      "1: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1           1\n",
      "2: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1           2\n",
      "3: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1           3\n",
      "4: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1           4\n",
      "5: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1 observation\n",
      "   segment_hex intensity_threshold\n",
      "1:        8000                  38\n",
      "2:          80                  38\n",
      "3:      800000                  38\n",
      "4:           0                  38\n",
      "5:  All_tissue                  NA\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m•\u001b[39m `` -> `...10`\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h[1] \"Upon loading randomization_datasheet, it has how many rows? 91\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"Calculating total numbers of missing explants per plate\"\n",
      "[1] \"Dimensions of missing explant data with totals\"\n",
      "[1]  0 14\n",
      "[1] \"Minimum number of explants on any plate after  removing missing and contaminated explants is...\"\n",
      "[1] Inf\n",
      "Empty data.table (0 rows and 14 cols): image_name,1,2,3,4,5...\n",
      "\n",
      "[1] \"About to add IDs to missing explant data based on filenames. Here is head of filenames.\"\n",
      "character(0)\n",
      "\n",
      "[1] \"We have added IDs to missing explant data based on filenames. Here is head of IDs.\"\n",
      "logical(0)\n",
      "\n",
      "[1] \"N rows of randomization datasheet before merging with missing explant data: \"\n",
      "[1] 91\n",
      "[1] \"Head of IDs in missing explant data\"\n",
      "logical(0)\n",
      "[1] \"Head of IDs in randomization datasheet\"\n",
      "[1] \"GTNEC1_1\" \"GTNEC1_2\" \"GTNEC1_3\" \"GTNEC1_4\" \"GTNEC1_5\" \"GTNEC1_6\"\n",
      "\n",
      "[1] \"Now about to merge randomization_datasheet and missing explant data.\"\n",
      "[1] \"Rows in randomization_datasheet: 91\"\n",
      "[1] \"Rows in missing explant data: 0\"\n",
      "\n",
      "[1] \"Now about to merge randomization datasheet  and missing explant data, by ID. Look at head of IDs for both.\"\n",
      "[1] \"GTNEC1_1\" \"GTNEC1_2\" \"GTNEC1_3\" \"GTNEC1_4\" \"GTNEC1_5\" \"GTNEC1_6\"\n",
      "logical(0)\n",
      "Error in bmerge(i, x, leftcols, rightcols, roll, rollends, nomatch, mult,  : \n",
      "  Incompatible join types: x.ID (logical) and i.ID (character)\n",
      "Calls: merge -> merge.data.table -> [ -> [.data.table -> bmerge\n",
      "In addition: Warning message:\n",
      "In min(na.omit(missing_explant_data_with_totals$total_explants)) :\n",
      "  no non-missing arguments to min; returning Inf\n",
      "Execution halted\n",
      "\u001b[?25hRscript grid_item_plots.R -d \"Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//\" -r \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/GTNEC_labels.xlsx\" -p 3 -v categorical -m 1 -M /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//output.csv -g 12 --samples-pre-labeling /mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//samples_pre_labeling.csv --sort 1 --height 5 --width 9 --Reporter Chl --outdir \"/mnt/output/gmodetector_out/\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25hWarning message:\n",
      "In storage.mode(default) <- type : NAs introduced by coercion\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h$datapath1\n",
      "[1] \"Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//\"\n",
      "\n",
      "$randomization_datasheet_path\n",
      "[1] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/GTNEC_labels.xlsx\"\n",
      "\n",
      "$`samples-pre-labeling`\n",
      "[1] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//samples_pre_labeling.csv\"\n",
      "\n",
      "$pixel_threshold\n",
      "[1] 3\n",
      "\n",
      "$variable\n",
      "[1] \"categorical\"\n",
      "\n",
      "$missing\n",
      "[1] 1\n",
      "\n",
      "$MissingList\n",
      "[1] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//output.csv\"\n",
      "\n",
      "$grid_type\n",
      "[1] 12\n",
      "\n",
      "$sort\n",
      "[1] 1\n",
      "\n",
      "$height\n",
      "[1] 5\n",
      "\n",
      "$width\n",
      "[1] 9\n",
      "\n",
      "$debug\n",
      "[1] 1\n",
      "\n",
      "$Reporter\n",
      "[1] \"Chl\"\n",
      "\n",
      "$keypath\n",
      "[1] \"FALSE\"\n",
      "\n",
      "$outdir\n",
      "[1] \"/mnt/output/gmodetector_out/\"\n",
      "\n",
      "$help\n",
      "[1] FALSE\n",
      "\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h[1] \"Saving list of input arguments to : /mnt/output/gmodetector_out//gmolabeler_stats_plots/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///Chl/gridplot_args.rds\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h[1] \"Reading in output from GMOlabeler at path: /mnt/output/gmodetector_out//gmolabeler_logic_outputs/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9//Chl/stats_with_sums_over_tissues.csv\"\n",
      "\u001b[?25h[1] \"Assuming default tissues for original poplar model because not specified with `keypath`\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"Rows in output from GMOlabeler: 6552\"\n",
      "\n",
      "[1] \"Max n_pixels_passing_threshold in output from GMOlabeler: 3675\"\n",
      "\n",
      "[1] \"Max total_signal in output from GMOlabeler: 428890.477472354\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"Look at the top of output from GMOlabeler\"\n",
      "[1] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[2] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[3] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[4] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[5] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "[6] \"/mnt/drives/Elements/Elements_25/GTNEC_GWAS_poplar_transformation_necrotic_test/day9///GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png\"\n",
      "\n",
      "[1] \"Look at the top of missing explant data\"\n",
      "\u001b[38;5;246m# A tibble: 0 × 3\u001b[39m\n",
      "\u001b[38;5;246m# ℹ 3 variables: filename <chr>, grid_item <chr>, present <chr>\u001b[39m\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"Look at the head of filenames for missing explant data\"\n",
      "character(0)\n",
      "\n",
      "[1] \"Look at the colnames of output from GMOlabeler\"\n",
      " [1] \"V1\"                         \"filename\"                  \n",
      " [3] \"grid_item\"                  \"segment_hex\"               \n",
      " [5] \"intensity_threshold\"        \"n_pixels_passing_threshold\"\n",
      " [7] \"mean_signal\"                \"max_signal\"                \n",
      " [9] \"total_signal\"               \"total_pixels\"              \n",
      "\n",
      "[1] \"Look at the colnames of missing explant data\"\n",
      "[1] \"filename\"  \"grid_item\" \"present\"  \n",
      "\n",
      "[1] \"Before merging the GMOlabeler and missing output data,they have rows of 6552 and 0 respectively.\"\n",
      "\u001b[?25h\u001b[?25h\n",
      "[1] \"After merging but before removing missing/contminated explants, merged output has 6552 rows.\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"After removing the missing/contaminated explants, merged output has 6552\"\n",
      "\u001b[?25h\n",
      "[1] \"After processing GMOlabeler output to merge with missing explantdata, Head of data (w/ first 5 col): \"\n",
      "                                      filename grid_item          V1\n",
      "1: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1           1\n",
      "2: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1           2\n",
      "3: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1           3\n",
      "4: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1           4\n",
      "5: GTENC1_I5.0_F1.9_L100LED30_100756_0_0_0.png         1 observation\n",
      "   segment_hex intensity_threshold\n",
      "1:        8000                  38\n",
      "2:          80                  38\n",
      "3:      800000                  38\n",
      "4:           0                  38\n",
      "5:  All_tissue                  NA\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m•\u001b[39m `` -> `...10`\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h[1] \"Upon loading randomization_datasheet, it has how many rows? 91\"\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "[1] \"Calculating total numbers of missing explants per plate\"\n",
      "[1] \"Dimensions of missing explant data with totals\"\n",
      "[1]  0 14\n",
      "[1] \"Minimum number of explants on any plate after  removing missing and contaminated explants is...\"\n",
      "[1] Inf\n",
      "Empty data.table (0 rows and 14 cols): image_name,1,2,3,4,5...\n",
      "\n",
      "[1] \"About to add IDs to missing explant data based on filenames. Here is head of filenames.\"\n",
      "character(0)\n",
      "\n",
      "[1] \"We have added IDs to missing explant data based on filenames. Here is head of IDs.\"\n",
      "logical(0)\n",
      "\n",
      "[1] \"N rows of randomization datasheet before merging with missing explant data: \"\n",
      "[1] 91\n",
      "[1] \"Head of IDs in missing explant data\"\n",
      "logical(0)\n",
      "[1] \"Head of IDs in randomization datasheet\"\n",
      "[1] \"GTNEC1_1\" \"GTNEC1_2\" \"GTNEC1_3\" \"GTNEC1_4\" \"GTNEC1_5\" \"GTNEC1_6\"\n",
      "\n",
      "[1] \"Now about to merge randomization_datasheet and missing explant data.\"\n",
      "[1] \"Rows in randomization_datasheet: 91\"\n",
      "[1] \"Rows in missing explant data: 0\"\n",
      "\n",
      "[1] \"Now about to merge randomization datasheet  and missing explant data, by ID. Look at head of IDs for both.\"\n",
      "[1] \"GTNEC1_1\" \"GTNEC1_2\" \"GTNEC1_3\" \"GTNEC1_4\" \"GTNEC1_5\" \"GTNEC1_6\"\n",
      "logical(0)\n",
      "Error in bmerge(i, x, leftcols, rightcols, roll, rollends, nomatch, mult,  : \n",
      "  Incompatible join types: x.ID (logical) and i.ID (character)\n",
      "Calls: merge -> merge.data.table -> [ -> [.data.table -> bmerge\n",
      "In addition: Warning message:\n",
      "In min(na.omit(missing_explant_data_with_totals$total_explants)) :\n",
      "  no non-missing arguments to min; returning Inf\n",
      "Execution halted\n",
      "\u001b[?25h\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd $gmolabeler_path\n",
    "for reporter in ${reporters[@]}; do\n",
    "    # Start the command with the base part\n",
    "    cmd=\"Rscript grid_item_plots.R \\\n",
    "    -d \\\"${data_folder}/\\\" \\\n",
    "    -r \\\"$randomization_datasheet\\\" \\\n",
    "    -p $pixel_threshold \\\n",
    "    -v categorical \\\n",
    "    -m 1 \\\n",
    "    -M $missing_explants \\\n",
    "    -g $grid \\\n",
    "    --samples-pre-labeling ${data}/samples_pre_labeling.csv \\\n",
    "    --sort 1 \\\n",
    "    --height $height \\\n",
    "    --width $width \\\n",
    "    --Reporter $reporter \\\n",
    "    --outdir \\\"${output_directory_prefix}\\\"\"\n",
    "\n",
    "    # Append the model key path if it's set and not None\n",
    "    if [ -n \"$segmentation_model_key\" ] && [ \"$segmentation_model_key\" != \"None\" ]; then\n",
    "        cmd+=\" --keypath \\\"$segmentation_model_key\\\"\"\n",
    "    fi\n",
    "\n",
    "    echo $cmd\n",
    "\n",
    "    # Execute the command\n",
    "    eval $cmd\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "echo -e \"Complete \\u2705\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  },
  "toc-autonumbering": true,
  "toc-showtags": false
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
